src/spec.ts

```diff
@@ -105,25 +124,31 @@ export function normalize(spec: ExtendedSpec): Spec {
     const hasRow = has(spec.encoding, ROW);
     const hasColumn = has(spec.encoding, COLUMN);
 
+    const isFacet = hasColumn || hasRow;
+
     // TODO: @arvind please  add interaction syntax here
     let encoding = duplicate(spec.encoding);
     delete encoding.column;
     delete encoding.row;
 
+    const data = extend(
+      { data: spec.data },
+      spec.transform ? { transform: spec.transform } : {}
+    );
+
     return extend(
       spec.name ? { name: spec.name } : {},
       spec.description ? { description: spec.description } : {},
-      { data: spec.data },
-      spec.transform ? { transform: spec.transform } : {},
+      isFacet ? {} : data,
       {
         facet: extend(
           hasRow ? { row: spec.encoding.row } : {},
           hasColumn ? { column: spec.encoding.column } : {}
         ),
-        spec: {
+        spec: extend(isFacet ? data : {}, {
```

Comment:
I think the correct thing is still putting data outside because we facets the data.  
Unless you have good reason why, let's not change this.